gateway_code/utils/tools.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding:utf-8 -*-
+
+# This file is a part of IoT-LAB gateway_code
+# Copyright (C) 2015 INRIA (Contact: admin@iot-lab.info)
+# Contributor(s) : see AUTHORS file
+#
+# This software is governed by the CeCILL license under French law
+# and abiding by the rules of distribution of free software.  You can  use,
+# modify and/ or redistribute the software under the terms of the CeCILL
+# license as circulated by CEA, CNRS and INRIA at the following URL
+# http://www.cecill.info.
+#
+# As a counterpart to the access to the source code and  rights to copy,
+# modify and redistribute granted by the license, users are provided only
+# with a limited warranty  and the software's author,  the holder of the
+# economic rights,  and the successive licensors  have only  limited
+# liability.
+#
+# The fact that you are presently reading this means that you have had
+# knowledge of the CeCILL license and that you accept its terms.
+
+""" common code for flash/debug tools """
+
+import logging
+import shlex
+
+from gateway_code.utils import subprocess_timeout
+
+LOGGER = logging.getLogger('gateway_code')
+
+
+class Tool(object):
+    """ common class for flash/debug tools """
+    TIMEOUT = 100
+
+    def __init__(self, verb=False, timeout=TIMEOUT):
+        self.verb = verb
+        self.timeout = timeout
+        self.name = self.__class__.__name__
+
+    def args(self, command_str): # pylint:disable=no-self-use
+        """splits a command into arguments for Popen"""
+
+        args = shlex.split(command_str)
+        return {'args': args, 'stderr': subprocess_timeout.STDOUT}
```

Comment:
The other problem with this is that the interactive output of the tool is lost. Example: in the integration tests, when testing a debug session with openocd, the output is displayed in real time by the process.
Getting stderr/stdout is already possible in the current state.

Reply:
yeah, I think in `subprocess.Popen(**self.args(cmd_debug))` for openocd we shouldn't use the common `args` method, or rewrite the stdout parameter before the Popen so that the interactive output is not lost, thanks